modules/DragonridingVigor.lua

```diff
@@ -53,18 +175,42 @@ function DragonridingVigor.prototype:EnteringWorld()
 end
 
 function DragonridingVigor.prototype:CheckShouldShow(event, unit, info)
+	if unit and unit ~= "player" then
+		return
+	end
+
+	-- Charges mode (11.2.7+)
+	if self.mode == "charges" or not self:UsingOldVigorWidgets() then
+		self.mode = "charges"
+		self.chargeSpellID = self.chargeSpellID or self:GetChargeSpell()
+
+		local show = self:ShouldShowCharges()
+		self:Show(show)
+
+		-- In charges mode we do NOT try to show/hide Blizzard's old widget frame
```

Comment:
I haven't been able to fire up the game to check, but: why not? Do both frames show in this case or is the Blizzard version being suppressed by another method?